src/agent/border_agent.cpp

```diff
@@ -191,6 +191,11 @@ void BorderAgent::HandleMdnsState(Mdns::Publisher::State aState)
     case Mdns::Publisher::State::kReady:
         UpdateMeshCopService();
         break;
+    case Mdns::Publisher::State::kRestarted:
+#if OTBR_ENABLE_SRP_ADVERTISING_PROXY
+        mAdvertisingProxy.PublishAllHostsAndServices();
```

Comment:
also call UpdateMeshCopService?

Reply:
My plan was to let the `Publisher` switch to `kReady` state after handling the `kRestarted` state.

Reply:
+1.